libraries/AP_Scripting/applets/leds_on_a_switch.lua

```diff
@@ -0,0 +1,45 @@
+-- leds_on_a_switch.lua: control led brightness with a radio switch
+--
+-- luacheck: only 0
+
+local rc_switch = rc:find_channel_for_option(300)  --AUX FUNC sw for mission restart
+
+if not rc_switch then  -- requires the switch to be assigned in order to run script
+    gcs:send_text(MAV_SEVERITY.INFO, "LED AUX channel not assigned")
+    return
+end
+
+-- bind a parameter to a variable
+function bind_param(name)
+    local p = Parameter()
+    assert(p:init(name), string.format('could not find %s parameter', name))
+    return p
+end
+
+-- constants
+local AuxSwitchPos = {LOW=0, MIDDLE=1, HIGH=2}
+local NTF_LED_BRIGHT        = bind_param("NTF_LED_BRIGHT")
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+-- state
+local prev_pos = -1
+
+function update()
+    local sw_pos = rc_switch:get_aux_switch_pos()
```

Comment:
If you use `get_aux_cached` it will work with aux functions over MAVLink too.

Reply:
Thanks, fixed